### PR TITLE
feat(core): separate social sign-in api

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Separate social sign-in API.

## Linear Issue Reference

LOG-2361
LOG-2362

## Testing

Passed UTs.

### CODE REVIEW

File: packages/core/src/routes/session.ts [hunk 1]
```diff
@@ -186,30 +186,43 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
   );
 
   router.post(
-    '/session/sign-in/social',
+    '/session/sign-in/social/send-authorization-request',
```
Comment (simeng-li) [inline packages/core/src/routes/session.ts:189]: /session/sign-in/social

File: packages/core/src/routes/session.ts [hunk 2]
```diff
@@ -186,30 +186,43 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
+      ctx.log(type, { redirectTo });
 
-        return next();
-      }
+      return next();
+    }
+  );
+
+  router.post(
+    '/session/sign-in/social/handle-authorization-callback',
```
Comment (simeng-li) [inline packages/core/src/routes/session.ts:214]: /session/sign-in/social/auth

File: packages/core/src/routes/session.ts [hunk 3]
```diff
@@ -186,30 +186,43 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
+    koaGuard({
+      body: object({
+        connectorId: string(),
+        code: string(),
+        redirectUri: string().regex(redirectUriRegEx),
```
Comment (simeng-li) [inline packages/core/src/routes/session.ts:219]: no need for redirectUri?

**Review by wangsijie — APPROVED** [review]
LGTM, better align with @simeng-li

File: packages/core/src/routes/session.ts [hunk 4]
```diff
@@ -190,26 +190,39 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
+      }),
+    }),
+    async (ctx, next) => {
+      const { connectorId, code, redirectUri } = ctx.guard.body;
+      const type = 'SignInSocial';
```
Comment (IceHe) [inline packages/core/src/routes/session.ts:224]: when split POST /session/sign-in/social into 

- POST /session/sign-in/social
- POST /session/sign-in/social/auth

it is not appropriate to reuse original log type `SignInSocial` for both of them.

File: packages/core/src/routes/session.ts [hunk 5]
```diff
@@ -190,26 +190,39 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
+      ctx.log(type, { connectorId, code, redirectUri });
```
Comment (IceHe) [inline packages/core/src/routes/session.ts:225]: After we split POST /session/sign-in/social into 

- POST /session/sign-in/social
- POST /session/sign-in/social/auth

I found that it isn't appropriate to use the log type `SignInSocial` (I added) for both of these situations.

Need further discussion.

- A. split log type too
- B. maybe do not log in POST /session/sign-in/social

**Pull request merged**